### PR TITLE
GH-1609 : Fix for MongoDB Atlas conn strings

### PR DESCRIPTION
**Pull request by ewassef**
Added recommended fix in the issue

# Description

Added a new format and a check in the server logic to format the connection string as recommended by MongoDB to handle `Cannot specify option 'loadBalanced' in TXT record`

## Issue reference
 

Please reference the issue this PR will close: #_[1609]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/mongodb/mongodb.go [hunk 1]
```diff
@@ -64,6 +64,9 @@ const (
 
 	// mongodb+srv://<server>/<params>
 	connectionURIFormatWithSrv = "mongodb+srv://%s/%s"
+
+	// mongodb+srv://<server>/<params>
```
Comment (ItalyPaleAle) [inline state/mongodb/mongodb.go:68]: Can you please update this comment too? Should be:

```
mongodb+srv://<username>:<password>@<server>/<params>
```

File: state/mongodb/mongodb.go [hunk 2]
```diff
@@ -355,6 +358,11 @@ func (m *MongoDB) Query(req *state.QueryRequest) (*state.QueryResponse, error) {
 
 func getMongoURI(metadata *mongoDBMetadata) string {
 	if len(metadata.server) != 0 {
```
Comment (ItalyPaleAle) [inline state/mongodb/mongodb.go:360]: Nit: there are now 4 different cases, with 4 different format strings. Perhaps we could clean up this a bit and streamline it?

**Review by ItalyPaleAle — APPROVED** [review]
LGTM! Only one minor thing. Please do fix the linter error however.

**Comment by ewassef** [comment]
Can someone re-run that failed step? It seems like a transient failure

**Comment by yaron2** [comment]
> Can someone re-run that failed step? It seems like a transient failure

See the lint failure here: (newline) https://github.com/dapr/components-contrib/pull/1698/files

File: state/mongodb/mongodb.go (hunk 1 above)
Comment (berndverst) [reply state/mongodb/mongodb.go:68]: Per @ItalyPaleAle 's comment - please update this comment

File: state/mongodb/mongodb.go (hunk 2 above)
Comment (ewassef) [reply state/mongodb/mongodb.go:360]: what do you have in mind? I think the variety is making the branching interesting here.

**Comment by ewassef** [comment]
@ItalyPaleAle updated as per your comments. 

FYI @yaron2

**Comment by berndverst** [comment]
@ewassef you forgot to sign-off on your last commit. Please make sure the DCO check passes.

**Comment by ewassef** [comment]
> @ewassef you forgot to sign-off on your last commit. Please make sure the DCO check passes.

done @berndverst

**Comment by yaron2** [comment]
> > @ewassef you forgot to sign-off on your last commit. Please make sure the DCO check passes.
> 
> done @berndverst

Did you run these commands? https://docs.dapr.io/contributing/contributing-overview/#i-didnt-sign-my-commit-now-what

**Comment by yaron2** [comment]
@ItalyPaleAle are you good with merging?

**Comment by ItalyPaleAle** [comment]
LGTM

**Review by yaron2 — APPROVED** [review]

**Comment by yaron2** [comment]
Thanks for the contribution @ewassef

**Pull request merged**